src/cpp/shared/game/game.cpp

```diff
@@ -529,4 +529,22 @@ void game::ProcessOnMapHeroes() {
       }
     }
   }
+}
+
+int getCastleOwnedIdx(playerData *player, int castleIdx) {
+  for (int i = 0; player->numCastles > i; ++i ) {
+    if (player->castlesOwned[i] == castleIdx )
+      return i;
+  }
+  return -1;
+}
+
+int game::getNumberOfThievesGuilds(int playerIdx) {
```

Comment:
Why were these functions imported?

Reply:
I could not manage to get these functions imported by changing asm

Reply:
Well, you've probably noticed that you would need to rename it in assembly to the proper mangled name. It's currently in there as game__getNumberOfThievesGuilds . I think you already did this.

The other part: It's currently declared "proc near C", which means that the name will be mangled to _game__getNumberOfThievesGuilds. If you change it to ?game@@getNumberOfThivesGuilds@@WhateverThisLooksLike@, the symbol would be changed to  _?game@@getNumberOfThivesGuilds@@WhateverThisLooksLike@, and hence VC++ would not find it.

So, you need to change that declaration to "proc near SYSCALL", which does no further mangling.

Our IDA-to-MASM script generates the declarations "proc near C" whenever it sees a name which isn't mangled in C++ form.